src/player_vs_bot.rs

```diff
@@ -1,19 +1,20 @@
 use anyhow::Result;
-use log::{error, info};
 use shakmaty::Move;
 
 use crate::{chess_engine::ChooseMove, chess_game::ChessGame};
 
 pub struct PlayerGame {
-    bot: Box<dyn ChooseMove + Send + Sync>,
+    bot: Box<dyn ChooseMove>,
```

Comment:
Kind of interesting how different player vs bot looks. I guess something more like
```rust
async fn player_vs_bot(game: Arc<Game>, player_moves_channel: WebsocketThingy<Move>, bot: Arc<dyn ChooseMove>) {
    info!("Player vs Bot Started...");

    loop {
        let player_move = player_moves_channel.recv().await;

        game.make_move(&player_move);

        let legal_moves = game.get_legal_moves();

        let bot_move = bot.choose_move(&game, &legal_moves);

        game.make_move(&bot_move);
    }
}
```

Would be less efficient because you would be using a websocket to recieve moves from the player (which I assume uses a socket and a tokio thread??), and a tokio blocking thread to await/ update the game state.

Where as doing this the more traditional http server route there is basically no blocking/ stalls and no need to use tokio threads to await move updates from the player. 

But pretty interesting how these code paths which do pretty similar things came out functionally pretty different

Reply:
I don't know if this is more or less efficient. Since this is async we could even use this approach in a regular tokio spawn. I used normal HTTP purely due to my familiarity with it

Reply:
I am guessing the only downside to this is it is more demanding of the client network